density_scaler/src/hex.rs

```diff
@@ -45,20 +45,29 @@ static HIP17_RES_CONFIG: [Option<HexResConfig>; 11] = [
     Some(HexResConfig::new(2, 1, 1)),     // 10
 ];
 
-#[derive(Debug, Serialize, Eq, PartialEq)]
-pub struct ScalingMap(HashMap<String, Decimal>);
+#[async_trait::async_trait]
+pub trait HexDensityMap: Clone {
+    async fn get(&self, hex: &str) -> Option<Decimal>;
+    async fn swap(&self, new_map: HashMap<String, Decimal>);
+}
+
+#[derive(Debug, Clone)]
+pub struct SharedHexDensityMap(Arc<RwLock<HashMap<String, Decimal>>>);
```

Comment:
If performance is a (the?) goal here, then using String for the map key when the actual key is an H3 cell is not going to help. Cells impl Hash directly. Cells also impl Ord and can be used in a BTreeMap (this is a case where I think a b-tree could make sense over a hash map)

Reply:
@jeffgrunewald

Reply:
> the location format over the wire that the verifier has

Got it. It's unfortunate that we don't keep indices `u64`, but given that constraint this makes sense.